docs/DeveloperGuide.md

```diff
@@ -1,34 +1,182 @@
+  * [UI component](DeveloperGuide.md#ui-component)
+  * [Logic component](DeveloperGuide.md#logic-component)
+  * [Model component](DeveloperGuide.md#model-component)
+  * [Storage component](DeveloperGuide.md#storage-component)
+  * [Common classes](DeveloperGuide.md#common-classes)
+* [Implementation](DeveloperGuide.md#implementation)
+* [Documentation](DeveloperGuide.md#documentation)
+* [Testing](DeveloperGuide.md#testing)
+* [Dev Ops](DeveloperGuide.md#dev-ops)
+* [Appendix](DeveloperGuide.md#appendix)
+  * [Appendix A: Product Scope](DeveloperGuide.md#appendix-a-product-scope)
+    * [Target user profile](DeveloperGuide.md#target-user-profile)
+    * [Value proposition](DeveloperGuide.md#value-proposition)
+  * [Appendix B: User Stories](DeveloperGuide.md#appendix-b-user-stories)
+  * [Appendix C: Use Cases](DeveloperGuide.md#appendix-c-use-cases)
+  * [Appendix D: Non Functional Requirements](DeveloperGuide.md#appendix-d-non-functional-requirements)
+  * [Appendix E: Glossary](DeveloperGuide.md#appendix-e-glossary)
+  * [Appendix F: Instructions for Manual Testing](DeveloperGuide.md#appendix-f-instructions-for-manual-testing)
+    * [Launch and Shutdown](DeveloperGuide.md#launch-and-shutdown)
+    * [Clearing the list](DeveloperGuide.md#clearing-the-list)
+    * [Saving data](DeveloperGuide.md#saving-data)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Setting up
 
+## Design
 
-## Product scope
-### Target user profile
+### Architecture
+
+### UI component
+
+The UI consists of `TextUi` class. It acts as a middleware between users' input
+and `Parser` class under **Logic Component**.
+
+The `UI` component,
+
+* Reads user commands using `Scanner` class.
+* Pass raw user commands into `Parser` class under **Logic Component**.
+* After the **Logic Component** has executed the function, it will then return `CommandOutput` object
+  which contains the outcome of the execution, fail or succeed.
+
+### Logic component
+
+![](images/LogicComponentStructure.png?raw=true "Structure of Logic Component")
+*Figure #. Structure of the Logic Component*
+
+**API** : [`seedu.duke.logic`](https://github.com/AY2021S2-CS2113T-T09-1/tp/tree/master/src/main/java/seedu/duke/logic) package (*TO BE CREATED LATER*)
+
+The Logic component,
+1. uses the `Parser` class to parse the user input obtained by the `TextUi` class of `UI`.
+2. This results in a specific `Command` object to be returned from the `Parser` class.
+3. Depending on the type of the `Command` object, its corresponding command execution will be called by `Duke` of `UI`.
+4. The command execution can affect the `Model` (e.g. clearing the list). 
+5. The result of the command execution is encapsulated as a `CommandOutput` object which is passed back to the `TextUi`.
+6. In addition, the `CommandOutput` object can also instruct the `TextUi` to perform certain actions, such as displaying the list to the user.
```

Comment:
Not clearly shown in the logic diagram, shouldn't the arrow point to the UI class since the CommandOutput calls the TextUi?